lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,523 @@
+            self.logger.error(f"Register User: missing username field")
+            abort(
+                400, message=f"missing username field",
+            )
+        if username.lower() == "admin":
+            self.logger.error(f"User tried to register with admin username")
+            abort(
+                400, message=f"Please choose another username",
+            )
+
+        password = user_data.get("password")
+        if not password:
+            self.logger.error(f"Register User: missing password field")
+            abort(
+                400, message=f"missing password field",
+            )
+
+        emailID = user_data.get("email")
+        if not emailID:
+            self.logger.error(f"Register User: missing email field")
+            abort(
+                400, message=f"missing email field",
+            )
+
+        firstName = user_data.get("firstName")
+        if not firstName:
+            self.logger.error(f"Register User: missing firstName field")
+            abort(
+                400, message=f"missing firstName field",
+            )
+
+        lastName = user_data.get("lastName")
+        if not lastName:
+            self.logger.error(f"Register User: missing lastName field")
+            abort(
+                400, message=f"missing lastName field",
+            )
+
+        # validate the email field
+        try:
+            valid = validate_email(emailID)
+            email = valid.email
+        except EmailNotValidError as e:
+            self.logger.error(f"Register User: invalid email {emailID}")
+            abort(400, message=f"Invalid email: {emailID}")
+
+        # check if user already exist
+        user = UserModel.query.filter_by(username=user_data.get("username")).first()
+        if not user:
+            try:
+                user = UserModel(
+                    bcrypt=self.bcrypt,
+                    bcrypt_log_rounds=self.bcrypt_log_rounds,
+                    username=username.lower(),
+                    password=password,
+                    firstName=firstName,
+                    lastName=lastName,
+                    email=email,
+                )
+
+                # insert the user
+                self.db_session.add(user)
+                self.db_session.commit()
+                self.logger.info(
+                    f"New user registered, Name: {firstName} {lastName} username: {username}, email: {email}"
+                )
+
+                # generate the auth token
+                auth_token = user.encode_auth_token(
+                    self.token_expire_duration, user.id, self.logger
+                )
+                if not auth_token:
+                    self.logger.info(
+                        f"User registered but unable to encode the auth token. username {username}"
+                    )
+                    return make_response(
+                        "{'message':'INTERNAL ERROR, unable to encode the auth token, please try logging in}",
```

Comment:
We don't want to tell the user what happened, just `abort(500, "INTERNAL ERROR")`